assets/dev/js/editor/document/elements/commands/settings.js

```diff
@@ -0,0 +1,229 @@
+
+			$e.run( 'document/elements/settings', {
+				container,
+				settings: isRedo ? changes.new : changes.old,
+				options: {
+					external: true,
+				},
+			} );
+		} );
+	}
+
+	/**
+	 * Function logHistory.
+	 *
+	 * Log history for settings command.
+	 *
+	 * @param {{}} args
+	 */
+	static logHistory( args, historyId = false ) {
+		const { containers = [ args.container ], settings = {}, isMultiSettings = false, options = {} } = args,
+			changes = {};
+
+		containers.forEach( ( container ) => {
+			const { id } = container,
+				newSettings = isMultiSettings ? settings[ container.id ] : settings;
+
+			if ( ! changes[ id ] ) {
+				changes[ id ] = {};
+			}
+
+			if ( ! changes[ id ].old ) {
+				changes[ id ] = {
+					old: {},
+					new: {},
+				};
+			}
+
+			Object.keys( newSettings ).forEach( ( settingKey ) => {
+				if ( 'undefined' !== typeof container.oldValues[ settingKey ] ) {
+					changes[ id ].old[ settingKey ] = elementorCommon.helpers.cloneObject( container.oldValues[ settingKey ] );
+					changes[ id ].new[ settingKey ] = newSettings[ settingKey ];
+				}
+			} );
+
+			delete container.oldValues;
+		} );
+
+		let historyItem = {
+			containers,
+			data: { changes },
+			type: 'change',
+			restore: Settings.restore,
+		};
+
+		if ( options.history ) {
+			historyItem = Object.assign( options.history, historyItem );
+		}
+
+		if ( historyId ) {
+			historyItem = Object.assign( { id: historyId }, historyItem );
+		}
+
+		$e.run( 'document/history/addSubItem', historyItem );
+	}
+
+	static getSubTitle( args ) {
+		const { containers = [ args.container ], settings = {}, isMultiSettings } = args,
+			settingsKeys = Object.keys( settings );
+
+		let result = '';
+
+		if ( ! isMultiSettings &&
+			1 === settingsKeys.length &&
+			containers[ 0 ].controls &&
+			containers[ 0 ].controls[ settingsKeys[ 0 ] ] ) {
+			result = containers[ 0 ].controls[ settingsKeys[ 0 ] ].label;
+		}
+
+		return result;
+	}
+
+	initialize() {
+		this.historyAction = false;
+	}
+
+	validateArgs( args ) {
+		this.requireContainer( args );
+
+		this.requireArgumentConstructor( 'settings', Object, args );
+	}
+
+	getArgsSnapshotId( args ) {
+		const { containers = [ args.container ], settings = {} } = args;
+
+		return containers.map( ( container ) => container.id ).join( ',' ) + ',' + Object.keys( settings );
+	}
+
+	getHistory( args ) {
+		if ( ! this.isHistoryActive() ) {
+			return false;
+		}
+
+		const { options = {} } = args,
+			currentArgsSnapshot = this.getArgsSnapshotId( args );
+
+		// Get current time.
+		const now = ( new Date() ).getTime();
+
+		if ( options.debounceHistory ) {
+			// If no timer or 1 second passed from last simulation.
+			if ( ! this.constructor.debounceTimer || now - this.constructor.debounceTimer > DEBOUNCE_DELAY ) {
+				this.historyAction = 'debounce';
+			}
+
+			// Clear all current snapshots
+			this.constructor.snapshots = this.constructor.snapshots.filter( ( snapshot ) => {
```

Comment:
Cannot use `currentArgsUniqueId` as key index for uniqueArgsStates since `getArgsUniqueId` does not return real unique id but unique id for the args each command.